pluggy/hooks.py

```diff
@@ -315,3 +315,9 @@ def __init__(self, plugin, plugin_name, function, hook_impl_opts):
         self.opts = hook_impl_opts
         self.plugin_name = plugin_name
         self.__dict__.update(hook_impl_opts)
+
+    def __repr__(self):
+        return "<HookImpl plugin_name=%r, plugin_file=%r>" % (
+            self.plugin_name,
+            self.plugin.__file__,
```

Comment:
wouldnt this completely fall appart for class instance based plugins?

Reply:
Fixed - using `repr()` now on the plugin.

As for the test: how would I test the non-instance based plugins?  (as used with tox for example)
btw: I've tried to use instance based plugins with tox (https://github.com/tarpas/pytest-testmon/pull/107), but it seems to have been just ignored then.